db-service/lib/cqn2sql.js

```diff
@@ -492,12 +494,25 @@ class CQN2SQLRenderer {
       return (this.sql = `INSERT INTO ${this.quote(entity)}${alias ? ' as ' + this.quote(alias) : ''} (${this.columns.map(c => this.quote(c))}) VALUES (${columns.map(param)})`)
     }
 
-    const extractions = this.managed(
-      columns.map(c => ({ name: c })),
-      elements,
-      !!q.UPSERT,
-    )
-    const extraction = extractions.map(c => c.sql)
+    const extractions = this.managed(columns.map(c => ({ name: c })), elements)
+    const extraction = extractions
+      .map(c => {
+        const element = elements?.[c.name]
+        if (element?.['@cds.extension']) {
+          return false
+        }
+        if (c.name === 'extensions__') {
+          const merges = extractions.filter(c => elements?.[c.name]?.['@cds.extension'])
+          if (merges.length) {
+            c.sql = `json_set(ifnull(${c.sql},'{}'),${merges.map(
+              c => this.string('$."' + c.name + '"') + ',' + c.sql,
+            )})`
+          }
+        }
+        return c
+      })
+      .filter(a => a)
+      .map(c => c.insert)
```

Comment:
Before this PR `c.insert` was `c.sql`?

Reply:
This PR changes the re-use function `managed` to better serve `INSERT`, `UPSERT` and `UPDATE`. As `UPSERT` requires both the `INSERT` and `UPDATE` behavior it now computes both inside `managed` and exposes all scenarios with their respective names.

This makes it also so that `managed` always generates the same results for any entity. Which makes it cacheable, but is not currently done.